### PR TITLE
Remove unneeded dependencies & misc. fixes.

### PR DESCRIPTION
**Pull request by eventualbuddha**
mocha, coffee-script, and assert were listed as runtime dependencies but they should be development dependencies. This PR also cleans up the npm test script and makes the tests pass.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "description": "Node module. typeOf('a string') == 'string', typeOf([]) == 'array', etc..",
   "main": "index.js",
   "scripts": {
-    "test": "./node_modules/mocha/bin/mocha -R spec --compilers coffee:coffee-script test.coffee"
+    "test": "mocha -R spec --compilers coffee:coffee-script test.coffee"
```
Comment (CodingFu) [inline package.json:7]: anyway, if i have no global installation of mocha i will not be able to run the tests, that is pretty bad.

File: test.coffee [hunk 2]
```diff
@@ -46,5 +46,4 @@ describe 'type of', ->
       constructor: ->
 
     obj = new ClassName()
-    asert.equal (typeOf obj), "classname"
-    
+    assert.equal (typeOf obj), "classname"
```
Comment (CodingFu) [inline test.coffee:49]: oops :smiley_cat:

**Comment by eventualbuddha** [comment]
You don't need to have a global installation of mocha. When NPM runs scripts it adds ./node_modules/.bin to your $PATH.

**Comment by CodingFu** [comment]
Didn't know about that feature.
Thanks a lot for merge request. You are merged! :)

**Pull request merged**